lib/compiler/src/sys_core_alias.erl

```diff
@@ -284,24 +284,31 @@ nodes_to_alias(Kind, Inner, Anno, Node, Keys0) ->
 %% Builds the key used to check if a value can be
 %% replaced by an alias. It considers literals,
 %% aliases, variables, tuples and cons recursively.
+%%
+%% We bail out after an arbitrary amount of nodes
+%% as very complex expressions may take forever to
+%% process otherwise.
+-define(DEPTH_LIMIT, 1000).
```

Comment:
Awesome, thank you!

I am wondering if the depth should be smaller. This pass requires a pattern to exist that matches the (sub)tree. If you have a large depth, it is unlikely that you have a pattern matching that tree. At the same tree, if you have a very deep tree and only a tiny subtree matches it, the cost of not allocating that subtree is very minor compared to allocating the rest of the tree. So I would say a depth of 100 should be enough for all practical purposes?

Reply:
> I am wondering if the depth should be smaller. [...] So I would say a depth of 100 should be enough for all practical purposes?

Good point, 1000 is far too cautious. I can't see any difference between 20 (!) and 1000 when compiling OTP, so 100 ought to be enough.

Reply:
I see! I wouldn't honestly mind making it 20 as well. If I reduce the reported tree to depth of 100, the pass still takes 50ms (which is still relatively high), while 20 is only 3ms. Although I guess none of those are really common enough to matter? :)